src/App.tsx

```diff
@@ -8,51 +8,109 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { getUserPosts } from './api/api';
+import { Post } from './types/Post';
 
 export const App: React.FC = () => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [visiblePosts, setVisiblePosts] = useState<Post[]>([]);
+  const [isCommentFormActive, setIsCommentFormActive] = useState(false);
+
+  const isVisiblePosts = !!visiblePosts.length && !isLoading && !isError;
+  const isNoPosts =
+    selectedUser && !visiblePosts.length && !isLoading && !isError;
+
+  const handleSelectUser = (user: User) => {
+    setSelectedUser(user);
+    setSelectedPost(null);
+  };
+
+  const handleSelectPost = (post: Post | null) => {
+    setSelectedPost(post);
+  };
+
+  const handleCommentForm = (value: boolean) => {
+    setIsCommentFormActive(value);
+  };
+
+  useEffect(() => {
+    if (selectedUser) {
+      setIsLoading(true);
+      getUserPosts(selectedUser.id)
+        .then(posts => {
+          setVisiblePosts(posts);
+          setIsLoading(false); // Встановлюємо isLoading в false після отримання постів
```

Comment:
Remove comments everywhere
```suggestion
          setIsLoading(false);
```